lib/internal/fs/promises.js

```diff
@@ -142,6 +146,26 @@ class FileHandle {
     this[kFd] = -1;
     return this[kHandle].close();
   }
+
+  [kTransfer]() {
+    const handle = this[kHandle];
+    this[kFd] = -1;
+    this[kHandle] = null;
+
+    return {
+      data: { handle },
+      deserializeInfo: 'internal/fs/promises:FileHandle'
+    };
+  }
+
+  [kTransferList]() {
+    return [ this[kHandle] ];
+  }
+
+  [kDeserialize]({ handle }) {
+    this[kHandle] = handle;
+    this[kFd] = handle.fd;
+  }
```

Comment:
@jasnell Happy to take suggestions on the API here. It’s a bit unfortunate that we can’t use the constructor and instead use this deserialization method to fill the object’s contents with the transferred data, but I don’t really know a way around this (See the comment in `JSTransferable::Data::Deserialize()` for details on why it seems necessary to me)

Reply:
This pattern is actually pretty common in JS for "hydrating" objects in my experience - though it is admittedly less neat than doing everything in the constructor and always getting a fully initialized object.

Reply:
I'm good with this API. The one thing that I would be concerned about is whether we'll always be certain that `kDeserialize` is always going to be called with a non-null/non-undefined value, otherwise the `({ handle })` deconstruction will fail.

Reply:
It will always be called with the value in `[kTransfer]().data`, so yes, that should work out.